### PR TITLE
Set errno for bpf_object__find_map_by_name

### PR DESCRIPTION
**Pull request by shpalani**
## Description
bpf_object__find_map_by_name():
 - Set errno to ENOENT.
 - Add a case to check errno set for invalid map name.
 - Added errno in doxygen comment.

## Testing

_Do any existing tests cover this change? No. 
Are new tests needed? Yes

## Documentation

_Is there any documentation impact for this change? No

## Installation

_Is there any installer impact for this change?No

### CODE REVIEW

File: include/bpf/libbpf.h [hunk 1]
```diff
@@ -247,6 +247,8 @@ bpf_object__close(struct bpf_object* object);
  * @param[in] name The name to look for.
  *
  * @returns The map found, or NULL if none.
+ *
+ * @exception ENOENT The map was not found.
```
Comment (dthaler) [inline include/bpf/libbpf.h:251]: These APIs are C apis and should never throw exceptions, so do not use `@exception`

Comment (shpalani) [reply include/bpf/libbpf.h:251]: doxygen does not have a way to comment errno. @exception is an alternative.

Comment (matthewige) [reply include/bpf/libbpf.h:251]: It looks like this is what the rest of the doxygen comments in this file uses. We should probably take a change to update the entire file to be consistent, though I am personally not sure exactly what tag we should use.

**Review by matthewige — APPROVED** [review]

**Review by shankarseal — APPROVED** [review]

File: include/bpf/libbpf.h (hunk 1 above)
Comment (shpalani) [reply include/bpf/libbpf.h:251]: Here is the tracking issue: https://github.com/microsoft/ebpf-for-windows/issues/3684

**Pull request merged**